File: 프로그래머스/3/43162. 네트워크/네트워크.py
from collections import deque

def bfs(start, visited, n, computers):
    queue = deque([start])
    visited[start] = True
    
    while queue:
        v = queue.popleft()
        for i in range(n):
            if computers[v][i] == 1 and not visited[i]:
                visited[i] = True
                queue.append(i)

def solution(n, computers):
    answer = 0
    visited = [False] * n
    
    for i in range(n):
        if not visited[i]:
            bfs(i, visited, n, computers)
            answer += 1
    return answer

File: 프로그래머스/3/43162. 네트워크/네트워크.java
import java.util.*;

class Solution {
    
    public void dfs(int depth, int[][] computers, int n, boolean[] visited) {
        visited[depth] = true;
        
        for (int i = 0; i < n; i++) {
            if (visited[i] == false && computers[depth][i] == 1) {
                dfs(i, computers, n, visited);
            }
        }
    }
    
    public int solution(int n, int[][] computers) {
        int answer = 0;
        boolean[] visited = new boolean[n];
        
        for (int i = 0; i < n; i++) {
            if (!visited[i]) {
                dfs(i, computers, n, visited);
                answer++;
            }
        }
        
        return answer;
    }
}